tests/0102-static_group_rebalance.c

```diff
@@ -0,0 +1,219 @@
+                    !parts || parts->cnt == 0) {
+                        if (parts)
+                                rd_kafka_topic_partition_list_destroy(parts);
+                        continue;
+                }
+
+                TEST_SAY("%s got assignment of %d partition(s)\n",
+                         rd_kafka_name(c->rk), parts->cnt);
+                rd_kafka_topic_partition_list_destroy(parts);
+                break;
+        }
+}
+
+static void rebalance_cb (rd_kafka_t *rk,
+                          rd_kafka_resp_err_t err,
+                          rd_kafka_topic_partition_list_t *parts,
+                          void *opaque) {
+
+        _consumer_t *c = opaque;
+
+        TEST_SAY("%s rebalance #%d/%d: %s: %d partition(s)\n",
+                 rd_kafka_name(rk),
+                 c->assign_cnt, c->max_rebalance_cnt,
+                 rd_kafka_err2name(err),
+                 parts->cnt);
+
+        switch (err)
+        {
+        case RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS:
+                c->assign_cnt++;
+
+                rd_kafka_assign(rk, parts);
+
+                TEST_ASSERT(c->assign_cnt <= c->max_rebalance_cnt,
+                            "%s rebalanced %d times, max was %d",
+                            rd_kafka_name(rk),
+                            c->assign_cnt, c->max_rebalance_cnt);
+                break;  
+
+        case RD_KAFKA_RESP_ERR__REVOKE_PARTITIONS:
+                rd_kafka_assign(rk, NULL);
+                break;
+
+        default:
+                TEST_FAIL("rebalance failed: %s", rd_kafka_err2str(err));
+                break;
+        }
+
+}
+
+#define _CONSUMER_CNT 2
+int main_0102_static_group_rebalance (int argc, char **argv) {
+        rd_kafka_conf_t *conf;
+
+        const int msgcnt = 100;
+        const char *topic = test_mk_topic_name("0102_static_group_rebalance", 1);
+        const char *subscription = tsprintf("^%s.*", topic);
+        _consumer_t c[_CONSUMER_CNT] = RD_ZERO_INIT;
+
+        test_conf_init(&conf, NULL, 60);
+
+        test_conf_set(conf, "session.timeout.ms", "5000");
+        test_conf_set(conf, "auto.offset.reset", "earliest");
+        test_conf_set(conf, "topic.metadata.refresh.interval.ms", "500");
+
+        rd_kafka_conf_set_opaque(conf, &c[0]);
+        test_conf_set(conf, "group.instance.id", "consumer1");
+        c[0].rk = test_create_consumer("static", &rebalance_cb,
+                                       rd_kafka_conf_dup(conf), NULL);
+
+        rd_kafka_conf_set_opaque(conf, &c[1]);
+        test_conf_set(conf, "group.instance.id", "consumer2");
+        c[1].rk = test_create_consumer("static", &rebalance_cb,
+                                       rd_kafka_conf_dup(conf), NULL);
+
+        c[0].max_rebalance_cnt = 1;
+        c[1].max_rebalance_cnt = 1;
+
+        test_consumer_subscribe(c[0].rk, subscription);
+        test_consumer_subscribe(c[1].rk, subscription);
+
+        test_create_topic(NULL, topic, 3, 1);
```

Comment:
Suggest you create and seed the topic before subscribing to make the test faster to run (otherwise subscribe will need to slowly poll metadata for the topic to become available)